### PR TITLE
1404 nc import letter file 1 case processor work

### PR DESCRIPTION
**Pull request by LukeLoze1**
***NOTE*** sort of blocked:
currently needs docker dev from https://github.com/ONSdigital/census-rm-docker-dev/pull/73 

# Motivation and Context
Add a Non Compliance Endpoint

# What has changed
New Endpoint

# How to test?
With the ATs off ticket.  
Cannot be merged before this ticket.

# Links
https://trello.com/c/iOTEEkUc/1381-nc-import-letter-file-1-bulk-processor-work-5

# Screenshots (if appropriate):

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/casesvc/messaging/RmNonComplianceReceiver.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    if (!StringUtils.isEmpty(collectionCase.getFieldOfficerId())) {
+      caze.setFieldOfficerId(collectionCase.getFieldOfficerId());
+    }
+
+    if (!StringUtils.isEmpty(collectionCase.getFieldCoordinatorId())) {
+      caze.setFieldCoordinatorId(collectionCase.getFieldCoordinatorId());
+    }
+
+    caseService.saveCaseAndEmitCaseUpdatedEvent(caze, null);
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/messaging/RmNonComplianceReceiver.java:49]: if the case was marked for fieldwork, it isn't sent to Fieldwork Adapter as a CREATE or whatever?

Comment (LukeLoze1) [reply src/main/java/uk/gov/ons/census/casesvc/messaging/RmNonComplianceReceiver.java:49]: I think this is covered by a followup ticket: https://trello.com/c/X9hDP5I3/1379-nc-import-file-2-and-send-cases-to-field-8

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/CaseMetadata.java [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
 
 import lombok.Data;
+import uk.gov.ons.census.casesvc.model.dto.NonComplianceTypeDTO;
 
 @Data
 public class CaseMetadata {
 
   private Boolean secureEstablishment;
   private String channel;
+  private NonComplianceTypeDTO nonCompliance;
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/CaseMetadata.java:11]: Noooo. Don't mix DTOs and entities!

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Breaks the cardinal rule of mixing DTO and entities - the DB needs to change independently of the DTOs, because case data stored in the DB would require patching if there's changes, and equally queues would need draining if DTOs got changed... all of which is horrible if the models are mixed and muddled

**Review by NickGrantONS — APPROVED** [review]
Looks good - my comments addressed

File: src/main/java/uk/gov/ons/census/casesvc/model/entity/EventType.java [hunk 3]
```diff
@@ -25,5 +25,6 @@
   DEACTIVATE_UAC,
   RM_CASE_UPDATED,
   RM_UNINVALIDATE_ADDRESS,
-  CLERICAL_ADDRESS_RESOLUTION
+  CLERICAL_ADDRESS_RESOLUTION,
+  SELECTED_FOR_NON_COMPLIANCE
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/model/entity/EventType.java:29]: All new events need to be added to Case API as well

**Review by hbrace — APPROVED** [review]

**Pull request merged**